server/src/server.rs

```diff
@@ -419,9 +419,19 @@ impl<B, L> Builder<B, L> {
 		self
 	}
 
-	/// Sets host filtering.
-	pub fn set_host_filtering(mut self, allow: AllowHosts) -> Self {
-		self.settings.allow_hosts = allow;
+	/// Enables host filtering and allow only the specified hosts.
+	///
+	/// Default: allow all.
+	pub fn host_filter<T: IntoIterator<Item = Authority>>(mut self, allow_only: T) -> Self {
```

Comment:
Just to copy my element comment:

It could accept anything that can `TryFrom` into the `Authority` type and then it'd have to return the error when `host_filter` is called if there is one. Prolly a bit nicer cos then you can just do

```rust
.host_filter([
    "foo.com",
    "something_else.com:*"
])?
```

for instance which would be quite cool.

(I think it should return an `AuthorityError` enum if the conversion fails personally rather than a `String`)